pkg/controller/accountclaim/accountclaim_controller.go

```diff
@@ -304,6 +304,8 @@ func (r *ReconcileAccountClaim) handleAccountClaimDeletion(reqLogger logr.Logger
 				reqLogger.Error(accountErr, "Failed updating account status for failed reuse")
 				return err
 			}
+
+			return err
```

Comment:
Now that we're _actually_ returning an error here, if `resetAccountSpecStatus` worked as expected but the claim wasn't cleaned up, we'll attempt to call this function again but the accountClaim.Spec.AccountLink will now be "", so we will never attempt to clean the account again and simply remove the finalizer

Reply:
Actually, scratch that, resetAccountSpecStatus doesn't affect the accountclaim, just the account.